debezium-core/src/main/java/io/debezium/jdbc/JdbcConnection.java

```diff
@@ -581,11 +582,49 @@ public synchronized boolean isConnected() throws SQLException {
     }
 
     public synchronized Connection connection() throws SQLException {
+        return connection(true);
```

Comment:
I think it'd be safer to not execute the statements when calling the parameterless `connection()`. Note that when using the PG connector, I see the insert being executed three times, so something is still not quite correct.

Reply:
@gunnarmorling It is correct, as postgres connector creates multiple connections in sequence. The functionality is not for executing DML (albeit I abuse it for tests) just mainly for session configuration and in this case I expect the session to be configured every time the connection is created.

Reply:
Ok, I'll expand the option's description a little bit then to clarify that fact.